### DISCUSSION TITLE
[@types/levelup] can't be used with the `levelup` package

### DISCUSSION THREAD

**Discussion by davidmurdoch** [DefinitelyTyped/DefinitelyTyped#59192] [Issues with a @types package]
The `levelup` package itself uses `module.exports = LevelUp;` but the @types/levelup package does `export const LevelUp: LevelUpConstructor`... which makes the types wrong and useless for those that use `levelup` directly.

Using `"levelup": "5.1.1"` with `"@types/levelup": "5.1.0"`, 

```
import leveldown from "leveldown";
import levelup from "levelup";
const store = leveldown("./test");
levelup(store) // TypeScript compilation error: "This expression is not callable"
// this is the correct JS usage of this library.
```

```
import leveldown from "leveldown";
import { LevelUp } from "levelup";
const store = leveldown("./test");
LevelUp(store) // Compiles, but now we get a Runtime error: "(0 , levelup_1.LevelUp) is not a function"
```

[a comment by an automated account is not shown]

**Comment by istherepie** [comment]
Thanks [davidmurdoch](https://github.com/davidmurdoch)!

You are correct, exported definitions do not mimic the original package. 
I'll have a look at how to fix it and get back to you!

**Comment by istherepie** [comment]
@davidmurdoch 

I have opened a pull request to resolve this issue: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59253

**Reply by istherepie** [reply]
The pull request #59253 has been merged.

**Comment by istherepie** [comment]
@davidmurdoch 

A new package of @types/levelup has been released on npm with version 5.1.1:

```
// package.json
  "dependencies": {
    ...
    "@types/levelup": "^5.1.1",
    ...
  }
```

This will now work as intended using the [export = and import = require() syntax](https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require):

```ts
import levelup = require("levelup");

// leveldown store
const leveldown = require("leveldown") ;
const store = leveldown("./testdb")

// levelup instance
const db = levelup(store)
```

Thanks again for report this and please mark this discussion as answered if you are satisfied with the fix!
